classes/author/maps/Schema.inc.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * @file classes/author/maps/Schema.inc.php
+ *
+ * Copyright (c) 2014-2020 Simon Fraser University
+ * Copyright (c) 2000-2020 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class author
+ *
+ * @brief Map authors to the properties defined in the announcement schema
+ */
+
+namespace PKP\author\maps;
+
+use APP\author\Author;
+use Illuminate\Support\Enumerable;
+use PKP\announcement\Announcement;
+use PKP\core\PKPApplication;
+use PKP\services\PKPSchemaService;
+
+class Schema extends \PKP\core\maps\Schema
+{
+    /** @var Enumerable */
+    public $collection;
+
+    /** @var string */
+    public $schema = PKPSchemaService::SCHEMA_AUTHOR;
+
+    /**
+     * Map an author
+     *
+     * Includes all properties in the announcement schema.
+     */
+    public function map(Author $item): array
+    {
+        return $this->mapByProperties($this->getProps(), $item);
+    }
+
+    /**
+     * Summarize an author
+     *
+     * Includes properties with the apiSummary flag in the author schema.
+     */
+    public function summarize(Author $item): array
+    {
+        return $this->mapByProperties($this->getSummaryProps(), $item);
+    }
+
+    /**
+     * Map a collection of Authors
+     *
+     * @see self::map
+     */
+    public function mapMany(Enumerable $collection): Enumerable
+    {
+        $this->collection = $collection;
+        return $collection->map(function ($item) {
+            return $this->map($item);
+        });
+    }
+
+    /**
+     * Summarize a collection of Authors
+     *
+     * @see self::summarize
+     */
+    public function summarizeMany(Enumerable $collection): Enumerable
+    {
+        $this->collection = $collection;
+        return $collection->map(function ($item) {
+            return $this->summarize($item);
+        });
+    }
+
+    /**
+     * Map schema properties of an Author to an assoc array
+     */
+    protected function mapByProperties(array $props, Author $item): array
+    {
+        $output = [];
+        foreach ($props as $prop) {
+            switch ($prop) {
+                // case '_href':
+                //     $output[$prop] = $this->getApiUrl('announcements/' . $item->getId());
+                //     break;
+                // case 'url':
+                //     $output[$prop] = $this->request->getDispatcher()->url(
+                //         $this->request,
+                //         PKPApplication::ROUTE_PAGE,
+                //         $this->context->getData('urlPath'),
+                //         'announcement',
+                //         'view',
+                //         $item->getId()
+                //     );
+                //     break;
```

Comment:
Remove commented out code.

Reply:
Here I am not sure about the `_href` property. Should it be removed also? if so, all the `switch` should be removed.